sample-compose/data/src/main/java/co/nimblehq/sample/compose/data/repository/AppPreferencesRepositoryImpl.kt

```diff
@@ -0,0 +1,40 @@
+package co.nimblehq.sample.compose.data.repository
+
+import android.util.Log
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.*
+import co.nimblehq.sample.compose.domain.repository.AppPreferencesRepository
+import kotlinx.coroutines.flow.*
+import java.io.IOException
+import javax.inject.Inject
+
+class AppPreferencesRepositoryImpl @Inject constructor(
+    private val appPreferencesDataStore: DataStore<Preferences>
+) : AppPreferencesRepository {
+
+    private object PreferencesKeys {
+        val FIRST_TIME_LAUNCH = booleanPreferencesKey("FIRST_TIME_LAUNCH")
+    }
+
+    override fun isFirstTimeLaunch(): Flow<Boolean> = appPreferencesDataStore.data
+        .catch { exception ->
+            if (exception is IOException) {
+                Log.i(
```

Comment:
Can we remove this or use `Timber` here?

Reply:
I believe we do not include `Timber` in the gradle of `data` layer @kaungkhantsoe 😄 
https://github.com/nimblehq/android-templates/blob/68d7608b5c2113a7450650a3a2a6b0cf77b6bf58/sample-compose/data/build.gradle.kts#L58-L83